### ISSUE TITLE
Default `resolverMainFields` should include "react-native"

### ISSUE THREAD

**Issue by kraenhansen** [react/metro#807]
**Do you want to request a *feature* or report a *bug*?**

It depends. I'd consider this a bug as there is an inconsistency between the Metro defaults and the actual values used when invoked from the React Native CLI.

**What is the current behavior?**

The current defaults for `resolver.resolverMainFields` doesn't include `"react-native"`:

```js
const {getDefaultConfig} = require('metro-config');
const defaults = getDefaultConfig.getDefaultValues();
console.log(defaults.resolver.resolverMainFields); // Logs [ 'browser', 'main' ]
```

I had a user of our library trying to extend the default metro config ending up in a situation where `"react-native"` was missing from the `resolverMainFields` because of this. See https://github.com/realm/realm-js/issues/4517#issuecomment-1106330586 for context.

**If the current behavior is a bug, please provide the steps to reproduce and a minimal repository on GitHub that we can `yarn install` and `yarn test`.**

See the reproduction above.

**What is the expected behavior?**

I'd expect this default to be the same as when the metro server is started through the React Native CLI: `['react-native', 'browser', 'main']`. See [cli-plugin-metro/src/tools/loadMetroConfig.ts#L85](https://github.com/react-native-community/cli/blob/master/packages/cli-plugin-metro/src/tools/loadMetroConfig.ts#L85) for the implementation.

**Please provide your exact Metro configuration and mention your Metro, node, yarn/npm version and operating system.**

N/A IMHO - ask and I'll provide this if you feel it's important for the description.

**Comment by motiz88** [comment]
Thanks for opening this issue @kraenhansen!

This works as intended - Metro is also used in non-React Native projects, so this would be a breaking change for them. I think the correct fix for https://github.com/realm/realm-js/issues/4517 is to not re-specify Metro's defaults in `metro.config.js` (i.e. don't spread `defaultResolver` at all). By omitting fields you don't want to override, you let Metro / the CLI take over with their defaults.

I think we can improve this with better documentation - i.e. the [configuration](https://facebook.github.io/metro/docs/configuration) page should at least mention that the React Native CLI introduces its own defaults for some options, and we can probably list them out in detail for each affected option.

**Comment by kraenhansen** [comment]
Fair enough :) I suppose I was tricked by the tag-line "The JavaScript bundler for React Native".

I've created a different PR with a suggested change to the docs: https://github.com/facebook/metro/pull/809

**Comment by robhogan** [comment]
Closing as resolved by the docs update - thanks!
